src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/SearchParameterOperations.cs

```diff
@@ -238,7 +238,7 @@ public async Task GetAndApplySearchParameterUpdates(CancellationToken cancellati
                 if (_searchParameterDefinitionManager.TryGetSearchParameter(searchParam.Uri, out var existingSearchParam))
                 {
                     // if the search parameter exists we should delete the old information currently stored
-                    DeleteSearchParameter(searchParam.Uri.ToString());
```

Comment:
>Uri.ToString() [](http://example.com/codeflow?start=54&length=14)

I've count 4 more non-tests places where we cast it to string.
Wonder what should we do in that places...

Reply:
SearchParameterStatusV1RowGenerator
SqlServerSearchParameterStatusDataStore
SearchParameterStatusCollection
CosmosDbSearchParameterStatusInitializer

Reply:
This is an unfortunate case where the programmer's default (ToString) produces unexpected behavior. The places you've called out should use OriginalString as well since that is what happens when we use ITypeElement. I'll update them.